src/agent/onefuzz-task/src/managed/cmd.rs

```diff
@@ -55,6 +96,10 @@ pub async fn run(args: &clap::ArgMatches) -> Result<()> {
             let err = format_err!("out of memory: {} bytes available, {} required", oom.available_bytes, oom.min_bytes);
             Err(err)
         },
+
+        _shutdown = shutdown_listener => {
```

Comment:
When we receive the `Shutdown` message, the `shutdown_listener` future will complete.